staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md

```diff
@@ -0,0 +1,106 @@
+# Minikube walkthrough
+
+This document will take you through setting up and trying the sample controller on a local minikube from a fresh clone of this repo.
+
+## Pre requisites
+
+- Go 1.7.x or later installed and setup. More infomration can be found at [go installation](https://golang.org/doc/install)
+- Dockerhub account to push the image to [Dockerhub](https://hub.docker.com/)
+
+## Install Minikube
+
+Minikube is a single node Kubernetes cluster that runs on your local machine. The Minikube docs have installation instructions for your OS.
+- [minikube installation](https://github.com/kubernetes/minikube#installation)
+
+## Clone the repository
+
+In order to build the sample controller image we will need to build the controller binary.
+
+```
+cd $GOPATH/src
+mkdir -p k8s.io
+cd k8s.io
+git clone https://github.com/kubernetes/sample-controller.git
+```
+
+## Build the binary
+
+Next we will want to create a new binary to both test we can build the controller and to use for the container image.
+
+From the root of this repo, where ```main.go``` is located, run the following command:
+```
+export GOOS=linux; go build .
+```
+if everything went well, you should have a binary called ```sample-controller``` present in your current directory.
+
+## Build the container image
+
+Using the binary we just built, we will now create a Docker image and push it to our Dockerhub registry so that we deploy it to our cluster.
+There is a sample ```Dockerfile``` located in ```artifacts/simple-image``` we will use this to build our own image.
+
+Again from the root of this repo run the following commands:
+```
+cp ./sample-controller ./artifacts/simple-image/kube-sample-controller
+docker build -t <YOUR_DOCKERHUB_USER>/kube-sample-controller:latest ./artifacts/simple-image
+docker push <YOUR_DOCKERHUB_USER>/kube-sample-controller
+```
+
+## Modify the replication controller
+
+You need to modify the [artifacts/examples/rc.yaml](/artifacts/examples/rc.yaml) file to change the ```imagePullPolicy``` to ```Always``` or ```IfNotPresent```.
+
+You also need to change the image from ```kube-sample-controller:latest``` to ```<YOUR_DOCKERHUB_USER>/kube-sample-controller:latest```. For example:
+
+```yaml
+...
+      containers:
+      - name: wardle-controller
+        image: <YOUR_DOCKERHUB_USER>/kube-sample-controller:latest
+        imagePullPolicy: Always
+...
+```
+
+Save this file and we are then ready to deploy and try out the sample controller.
+
+## Deploy to Minikube
+
+We will need to create several objects in order to setup the sample controller so you will need to ensure you have the ```kubectl``` tool installed. [Install kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+```
```

Comment:
Can we just do a `kubectl apply -f ./`?

If we care about the order, we can merge the order sensitive objects in one yaml file and separate them by `===`.

Reply:
I have merged them into one yaml `sample-controller.yaml`